tests/MongoDB.Driver.Core.Tests/Core/Connections/BinaryConnectionTests.cs

```diff
@@ -276,6 +276,45 @@ public void Dispose_should_raise_the_correct_events()
             act.ShouldThrow<InvalidOperationException>();
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void ReceiveMessage_should_throw_error_when_message_is_an_invalid_size(
```

Comment:
This is totally optional, but a slightly more modern take might be to declare the test as:
`public async void` and use the `await` keyword for the `async` variant. See [this test](https://github.com/mongodb/mongo-csharp-driver/blob/7d426a8243532983aee819d2b227c873d80df75d/tests/MongoDB.Driver.Tests/PinnedShardRouterTests.cs#L55) for an example. Completely optional, as the way you have it here definitely works, so we could save it for next time 😄

Reply:
Either way there is a mismatch because we are using a single test to test both the sync and async variants of a method.

Either:

1. we are using GetAwaiter().GetResult() when testing the async method
2. we are blocking in an async test method when testing the sync method (blocking is bad in async code)

In production code both those scenarios would not be tolerated. In test code I don't see much advantage to using XUnit's new support for async test methods unless the test itself is 100% async.

If you can think of reasons why the new async test method is better, elaborate (other than being newer!).

Reply:
Hmm, you make a good point: I agree that both scenarios would not be tolerated in production code, so in that sense they're both "equally unacceptable."

Initially, my thought process was that they're both equally unacceptable, then I'd prefer to use the version that's slightly newer and results in slightly more concise code. 

As a counterpoint to my own preference for `async void`, before C#7 introduced `async main`, I think standard practice was to call use define an async helper and to call the helper with `GetAwaiter.GetResult()`, so there's at least an accepted history behind using `GetAwaiter.GetResult()` in a synchronous method.

Just as a discussion point, as an alternative, I suppose we could have a sync and async version of each test defined, with a helper method returning Assert, Arrange, Act Actions, but I'm not sure we'd gain much over the current implementation.

TLDR: No action needed, resolving this thread :)